### DISCUSSION TITLE
To do task due today

### DISCUSSION THREAD

**Discussion by jorgemarmor** [RogerSelwyn/O365-HomeAssistant#257] [Q&A]
Hi everyone,

I've been using for a while the following code to create toDo tasks that are due the day they are created, but they are not working anymore:

action: o365.new_todo
metadata: {}
data:
  due: "{{ now() }}"
  subject: "xxxx*
target:
  entity_id: todo.nuestro_pisito_account1

They were working but they are not working anymore. I get the following error:
![Screenshot_2024-10-05-18-16-59-440_io homeassistant companion android-edit](https://github.com/user-attachments/assets/a39a4497-dfc6-495e-b81f-f4cd69cc73ca)

Any idea what's is wrong now?

Thanks in advance

**Comment by RogerSelwyn** [comment]
I guess the obvious question is, what have you changed? Have you upgraded HA or the O365 integration?

**Comment by RogerSelwyn** [comment]
Then next step try creating a todo through the developer -> actions tab, and report if that works.

**Comment by jorgemarmor** [comment]
That's quick answering 🤣

I don't know for how long it has stopped working, but definitely I've been
updating both HA and O365.

Creating a task with a set due date works. It just the traditional now()
formula not working anymore.

**Comment by RogerSelwyn** [comment]
Then I would suggest that your formula doesn’t work because HA core has changed something, which I can’t help with. Make sure a valid date is sent and it should work (as you have proved).

**Comment by jorgemarmor** [comment]
Ok, I will try.

**Comment by jorgemarmor** [comment]
I couldn't find if it is due to a change in the core but I managed to fix it by using now().strftime('%Y-%m-%d').
